web/src/test/java/org/springframework/security/web/server/authentication/SwitchUserWebFilterTests.java

```diff
@@ -0,0 +1,463 @@
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.mock.http.server.reactive.MockServerHttpRequest;
+import org.springframework.mock.web.server.MockServerWebExchange;
+import org.springframework.security.authentication.AccountStatusUserDetailsChecker;
+import org.springframework.security.authentication.AuthenticationCredentialsNotFoundException;
+import org.springframework.security.authentication.DisabledException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextImpl;
+import org.springframework.security.core.userdetails.ReactiveUserDetailsService;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.switchuser.SwitchUserGrantedAuthority;
+import org.springframework.security.web.server.WebFilterExchange;
+import org.springframework.security.web.server.context.ServerSecurityContextRepository;
+import org.springframework.security.web.server.context.WebSessionServerSecurityContextRepository;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.web.server.WebFilterChain;
+import reactor.core.publisher.Mono;
+
+import java.security.Principal;
+import java.util.Collections;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+import static org.springframework.security.core.context.ReactiveSecurityContextHolder.withSecurityContext;
+import static org.springframework.security.web.server.authentication.SwitchUserWebFilter.ROLE_PREVIOUS_ADMINISTRATOR;
+
+/**
+ * @author Artur Otrzonsek
+ * @since 5.2.6
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class SwitchUserWebFilterTests {
+
+	private SwitchUserWebFilter switchUserWebFilter;
+
+	@Mock
+	private ReactiveUserDetailsService userDetailsService;
+	@Mock
+	ServerAuthenticationSuccessHandler successHandler;
+	@Mock
+	private ServerAuthenticationFailureHandler failureHandler;
+	@Mock
+	private ServerSecurityContextRepository serverSecurityContextRepository;
+
+	@Rule
+	public ExpectedException exceptionRule = ExpectedException.none();
+
+	@Before
+	public void setUp() {
+		switchUserWebFilter = new SwitchUserWebFilter(userDetailsService, successHandler, failureHandler);
+		switchUserWebFilter.setSecurityContextRepository(serverSecurityContextRepository);
+	}
+
+	@Test
+	public void switchUSerShouldDoNothingIfRequestNotMatch() {
```

Comment:
In an effort to have test names be consistent across the codebase, we would like all new tests to follow the naming convention `methodWhenConditionThenResult`.
For example, this test could be renamed to `switchUserWhenRequestNotMatchThenDoesNothing`.

Reply:
done